File: 프로그래머스/0/120843. 공 던지기/공 던지기.cs
using System;

public class Solution 
{
    public int solution(int[] numbers, int k) 
    {
        int count = 1;
        int index = 1;
        while(true)
        {
            count++;
            index += 2;
            if(index > numbers[numbers.Length-1])
            {
                switch(index - numbers[numbers.Length-1])
                {
                    case 1:
                        index = 1;
                        break;
                    case 2:
                        index = 2;
                        break;
                }
            }
            if(count == k)
            {
                break;
            }
        }
        return numbers[index-1];
    }
}

File: 프로그래머스/0/120843. 공 던지기/공 던지기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> numbers, int k) 
{
    int index = 0;
    int count = 1;
    while(count != k)
    {
        index += 2;
        if(index > numbers.size()-1)
        {
            index = index - numbers.size();
        }
        count++;
    }
    return numbers[index];
}